### PR TITLE
[Jormun] Allow more fallback combinations

### PR DESCRIPTION
**Pull request by xlqian**

### CODE REVIEW

File: source/jormungandr/jormungandr/fallback_modes.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from enum import Enum
+
+
+class FallbackModes(Enum):
+    walking = 0
```
Comment (nonifier) [inline source/jormungandr/jormungandr/fallback_modes.py]: Why not using the modes already defined in `StreetNetworkMode` (request.proto) ? You have available :
```python
    response_pb2.Walking,
    response_pb2.Bike,
    response_pb2.Bss,
    response_pb2.Car,
    response_pb2.Ridesharing,
    response_pb2.CarNoPark,
```

File: source/jormungandr/jormungandr/fallback_modes.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        # In principle, non personal modes are walking, bss, taxi and ridesharing, since we want to minimize the number
+        # of ridesharing, we handle combinations with ridesharing manually
+        # personal modes are bike and car, we don't allow to use personal modes in ending fallbacks, except bike bike.
+        non_personal_modes = {cls.walking, cls.bss, cls.taxi}
+        return (
+            _combi(cls.modes_enum() - {cls.ridesharing}, non_personal_modes)
+            # we remove bss walking and  walking bss, since they are redundant for kraken (bss includes walking)
+            - {(cls.walking, cls.bss), (cls.bss, cls.walking)}
+        ) | {
+            (cls.bike, cls.bike),
```
Comment (nonifier) [inline source/jormungandr/jormungandr/fallback_modes.py]: A little comment would be appreciated about why we can mix `bike` (a personal mode) with PT.

File: source/jormungandr/jormungandr/fallback_modes.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    ridesharing = 4
+    taxi = 5
+
+    @classmethod
+    def modes_str(cls):
+        return {e.name for e in cls}
+
+    @classmethod
+    def modes_enum(cls):
+        return set(cls)
```
Comment (nonifier) [inline source/jormungandr/jormungandr/fallback_modes.py:48]: 🧙‍♂️ :)

**Review by nonifier — APPROVED** [review]

File: source/jormungandr/jormungandr/fallback_modes.py (hunk 3 above)
Comment (xlqian) [reply source/jormungandr/jormungandr/fallback_modes.py:48]: its-magic

**Comment by nonifier** [comment]
When your dependency script reject your PR ! 
#winning

**Review by AurelienLP — APPROVED** [review]

**Pull request merged**